### ISSUE TITLE
Woocommerce doesnt allow structured data for product type to be an array

### ISSUE THREAD

**Issue by erikdemarco** [woocommerce/woocommerce#30964]
If you look at this guidelines for product schema:
https://developers.google.com/search/docs/advanced/structured-data/product

It stated like this:
isbn is only a valid property on Book. To use this correctly, you will need to co-type the Product. This will let you use properties of both types on the node. For example: 
```
{
  "@context": "https://schema.org",
  "@type": ["Product", "Book"],
  ....
}
```

So if we provide isbn to product schema and change the product type to ["Product", "Book"]
Woocommerce will gives error: 
`preg_match() expects parameter 2 to be string, array given: wp-content/plugins/woocommerce/includes/class-wc-structured-data.php:47`

Here's the example code:
```
add_filter('woocommerce_structured_data_product', 'woocommerce_structured_data_product_add_custom_schema', 10, 2);
function woocommerce_structured_data_product_add_custom_schema($markup, $product){
    $markup['@type'] = [ 'Book', 'Product' ];
    return $markup;
}
```

**Comment by juliaamosova** [comment]
Hi @erikdemarco,

Thank you for opening the issue! It requires further feedback from the WooCommerce Core team. I am adding the `needs developer feedback` label to this issue so that the Core team could take a look. 

Please note it may take a few days for them to get to this issue. Thank you for your patience.

**Comment by barryhughes** [comment]
Agree, looks like we may be able to solve with a small tweak to `WC_Structured_Data::set_data()` (plus any supporting tests).

**Comment by kkoteen** [comment]
Hi @erikdemarco - thanks for the report.

I know this has been open for a very long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I've taken a look and marked this ready for the engineering team to pick up.

A few related open issues worth linking here, all touching the same structured data implementation:

[#25969](https://github.com/woocommerce/woocommerce/issues/25969) — Grouped product schema outputs an incorrect single price instead of a price range
[#41060](https://github.com/woocommerce/woocommerce/issues/41060) — Shipping details (shippingDetails, shippingDestination) not included in product schema, despite Google supporting this since 2023
[#48401](https://github.com/woocommerce/woocommerce/issues/48401) — Organization-level return policy markup not reflected in structured data output
All four point to the same underlying gap — our structured data implementation hasn't kept pace with Google's schema spec. Would be worth tackling these together.

What I'm seeing: Current trunk still requires structured-data `@type` to be a single alphabetic string: `WC_Structured_Data::set_data()` passes `$data['@type']` straight into `preg_match()`, and `get_structured_data()` later lowercases that value as a scalar key. The array-typed `@type` customization shown in the issue is therefore still unsupported on trunk, and no matching open PR was found in local PR metadata.
